### DISCUSSION TITLE
Some buggy behavior in v1.12.1

### DISCUSSION THREAD

**Discussion by iiz00** [vslinko/obsidian-outliner#151] [Archive]
In v1.12.1, I'm experiencing some buggy behavior.

1. Typing Enter at an empty bullet point doesn't outdent, but  deletes the bullet point (and breaks the line).

2. Tab key in the middle of the line doesn't indent, but make spaces (Obsidian's default behavior). 

3. Sometimes BackSpace key and Delete key also behave as default.

4. (This may be specific to Japanese IME and a different kind of problem from 1~3.) When we  input Japanese, we type hiragana and then convert them into kanji. To complete the conversion, we have to type Enter key. In previous version, the first Enter finished the conversion and second one made line breaks. But in 1.12.1 (I don't know how it was in 1.12.0), the first Enter to finish conversion breaks line at the same time.It seems to occur only when "Enhance the Enter key" option is on.

All of the above occur not always, but frequently. 
Though I can avoid 4 by disabling "Enhance the Enter key" option, I saw others occur even if I disabled my Japanese IME.

As a workaround, downgrading to v1.1.11 (I have not tried 1.1.12 and 1.2.0) seems to have resolved the problems.
Do you have any idea to fix them?

**Comment by vslinko** [comment]
@iiz00 Could you please provide some example file which has problems?

**Comment by iiz00** [comment]
I'm sorry, but I have made some mistakes in the previous post.
No.4 (Typing `Enter` for hiragana to kanji composition causes an unwanted line break ) is the main problem. 
By disabling `Enhance the Enter key` option I can avoid the problem with the line break, but then  
`Enter` at an empty line doesn't make an outdent.
I misunderstood this behavior as problem No.1, but it may be an intended behavior when `Enhance the Enter key`option is disabled.
As for No.3, I might have disabled `Stick the cursor to the content`option.
I'm not sure about `Tab` behavior of No.2, but now I couldn't reproduce the problem.

No.4 is already mentioned by @yo-goto at issue #154.
This always happens when `Enhance the Enter key`is enabled in v1.2.1 or v1.2.5.
In v1.1.11 or earlier, this kind of problem doen't happen even if the option is enabled.
So I think some changes in processing key input in v1.2 caused this problem.
